core/src/main/java/org/jruby/ir/runtime/IRReturnJump.java

```diff
@@ -1,19 +1,21 @@
 package org.jruby.ir.runtime;
 
 import org.jruby.exceptions.Unrescuable;
-import org.jruby.runtime.DynamicScope;
+import org.jruby.ir.IRScope;
 
 public class IRReturnJump extends IRJump implements Unrescuable {
-    public final DynamicScope methodToReturnFrom;
+    public final IRScope returnScope;
+    public final IRScope methodToReturnFrom;
     public final Object returnValue;
 
-    private IRReturnJump(DynamicScope scopeToReturnFrom, Object rv) {
+    private IRReturnJump(IRScope returnScope, IRScope scopeToReturnFrom, Object rv) {
```

Comment:
Naming is a little weird here...these mean the same thing in my head.

Reply:
returnScope is the scope where the return lives and scopeToReturnFrmo is where the return will exit from.  I am happy to change those though.

Reply:
Maybe sourceScope and targetScope, or returnScope and targetScope? Entirely up to you, it's a minor issue.